### DISCUSSION TITLE
RenderModeProvider & ActiveCircuitState services

### DISCUSSION THREAD

**Discussion by ecaldentey** [MarimerLLC/csla#3992] [Questions]
Hello Rocky,

In your implementation of ProjectTracker for .NET 8 in the Blazor client project, in the Program.cs class, you register the following services which are specific to the ProjectTracker project but not part of the CSLA framework. Can you explain their utility and whether it is necessary to create these services in my client project?

// Add render mode detection services
builder.Services.AddTransient<RenderModeProvider>();
builder.Services.AddScoped<ActiveCircuitState>();

Thanks in advance

**Comment by rockfordlhotka** [comment]
https://blog.lhotka.net/2024/03/30/Blazor-8-Render-Mode-Detection
